src/wallet/wallet.cpp

```diff
@@ -704,18 +704,24 @@ void CWallet::SetUsedDestinationState(WalletBatch& batch, const uint256& hash, u
 {
     AssertLockHeld(cs_wallet);
     const CWalletTx* srctx = GetWalletTx(hash);
+    std::set<CTxDestination> tx_destinations;
     if (!srctx) return;
 
     CTxDestination dst;
     if (ExtractDestination(srctx->tx->vout[n].scriptPubKey, dst)) {
         if (IsMine(dst)) {
             if (used && !GetDestData(dst, "used", nullptr)) {
-                AddDestData(batch, dst, "used", "p"); // p for "present", opposite of absent (null)
+                if (AddDestData(batch, dst, "used", "p")) // p for "present", opposite of absent (null)
+                    tx_destinations.insert(dst);
             } else if (!used && GetDestData(dst, "used", nullptr)) {
                 EraseDestData(batch, dst, "used");
             }
         }
     }
+
+    // Mark all other outputs to the same destination dirty so
```

Comment:
Move this comment to the function declaration in the header

Reply:
done, also reworded it to be more general